jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/NativeLibLoader.java

```diff
@@ -34,21 +39,29 @@
 class NativeLibLoader {
   private static final Log logger = LogFactory.getLog(NativeLibLoader.class);
 
+  private static Path mappedFilesBaseDir = Paths.get("/proc/self/map_files");
+
   /**
    * Supported OS enum.
    */
   enum OS {
     WINDOWS("windows"),
     MACOS("macos"),
     LINUX("linux"),
+    LINUX_MUSL("linux-musl"),
     SOLARIS("solaris");
 
     final String name;
 
-    private OS(String name) {
+    OS(String name) {
       this.name = name;
     }
 
+    @VisibleForTesting
```

Comment:
Is this annotation necessary? It is package private, so it should be visible for the test code already. There aren't any other usages of guava in the xgboost4j library so this now makes it a runtime dependency whereas before it was optional.

Reply:
I've added it as an indication for the reader why this method has relaxed visibility - or in this case even, why it even exists (it's only used by the test). It's not required to execute test though. I can remove it.

REMOVED (303bfcda)

Reply:
I understand the desire to document it, but in this case it's probably better to just leave a comment rather than pull in guava.